PetaPoco.Tests.Integration/Databases/MySQLConnector/MySqlConnectorQueryTests.cs

```diff
@@ -9,5 +15,169 @@ public MySqlConnectorQueryTests()
+                         ORDER BY 1 DESC
+                         LIMIT 1;";
+
+            List<Order> result;
+            using (var multi = DB.QueryMultiple(sql, "Peta0"))
+            {
+                result = multi.Read<Order, Person, Order>((o, p) =>
+                {
+                    o.Person = p;
+                    return o;
+                }).ToList();
+            }
+
+            result.Count.ShouldBe(1);
+
+            var order = result.First();
+
+            order.PoNumber.ShouldStartWith("PO");
+            order.Status.ShouldBeOneOf(Enum.GetValues(typeof(OrderStatus)).Cast<OrderStatus>().ToArray());
+            order.PersonId.ShouldNotBe(Guid.Empty);
+            order.CreatedOn.ShouldBeLessThanOrEqualTo(new DateTime(1990, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            order.CreatedBy.ShouldStartWith("Harry");
+
+            order.Person.ShouldNotBeNull();
+            order.Person.Id.ShouldNotBe(Guid.Empty);
+            order.Person.Name.ShouldStartWith("Peta");
+            order.Person.Age.ShouldBe(18);
+        }
+
+        [Fact]
+        public override void QueryMultiple_ForMultiResultsSetWithSinglePoco_ShouldReturnValidPocoCollection()
+        {
+            AddOrders(1);
+
+            var pd = PocoData.ForType(typeof(Person), DB.DefaultMapper);
+            var od = PocoData.ForType(typeof(Order), DB.DefaultMapper);
+            var pdName = pd.Columns.Values.First(c => c.PropertyInfo.Name == "Name").ColumnName;
+            var odId = od.Columns.Values.First(c => c.PropertyInfo.Name == "Id").ColumnName;
+
+            var sql = $@"SELECT * FROM {DB.Provider.EscapeTableName(od.TableInfo.TableName)} o
+                         WHERE o.{DB.Provider.EscapeSqlIdentifier(odId)} = @0;
+                         SELECT * FROM {DB.Provider.EscapeTableName(pd.TableInfo.TableName)} p
+                         WHERE p.{DB.Provider.EscapeSqlIdentifier(pdName)} = @1;";
+
+            Order order;
+            using (var multi = DB.QueryMultiple(sql, "1", "Peta0"))
+            {
+                order = multi.Read<Order>().First();
+                order.Person = multi.Read<Person>().First();
+            }
+
+            order.PoNumber.ShouldStartWith("PO");
+            order.Status.ShouldBeOneOf(Enum.GetValues(typeof(OrderStatus)).Cast<OrderStatus>().ToArray());
+            order.PersonId.ShouldNotBe(Guid.Empty);
+            order.CreatedOn.ShouldBeLessThanOrEqualTo(new DateTime(1990, 1, 1, 0, 0, 0, DateTimeKind.Utc));
+            order.CreatedBy.ShouldStartWith("Harry");
+
+            order.Person.ShouldNotBeNull();
+            order.Person.Id.ShouldNotBe(Guid.Empty);
+            order.Person.Name.ShouldStartWith("Peta");
+            order.Person.Age.ShouldBe(18);
+        }
+
+        // FIXME: firstOrderLine.Quantity.ToString() should be "1" but was "2"
+        [Fact]
+        public override void QueryMultiple_ForMultiResultsSetWithMultiPoco_ShouldReturnValidPocoCollection()
```

Comment:
Seems this should be obvious, but I'm missing what's wrong here. Same assertion failure manifesting on all MySql-based dbms and drivers (MySql, MySql Connector, MariaDB)

Reply:
Consolidating this thread to the [MySQL](https://github.com/CollaboratingPlatypus/PetaPoco/pull/696#discussion_r1338988806) comments since they stem from the same issue.

Reply:
[Resolved](https://github.com/CollaboratingPlatypus/PetaPoco/pull/696#discussion_r1339138316) with same fix as MySql.